crates/compiler/fmt/src/lib.rs

```diff
@@ -69,16 +69,25 @@ impl<'a> Buf<'a> {
     }
 
     pub fn push_str_allow_spaces(&mut self, s: &str) {
-        debug_assert!(!self.beginning_of_line);
+        debug_assert!(
+            !self.beginning_of_line,
+            "push_str: `{s}` with text:\n{}",
+            self.text
+        );
 
         self.flush_spaces();
 
         self.text.push_str(s);
     }
 
     pub fn push_str(&mut self, s: &str) {
-        debug_assert!(!self.beginning_of_line);
-        debug_assert!(!s.contains('\n') && !s.ends_with(' '));
+        debug_assert!(
+            !self.beginning_of_line,
+            "push_str: `{s}` with text:\n{}",
+            self.text
+        );
+        debug_assert!(!s.contains('\n'));
+        debug_assert!(!s.ends_with(' '));
```

Comment:
having this be separate asserts means that when one fails it's one step quicker to figure out what went wrong